### ISSUE TITLE
Script fails to run on Mac OS

### ISSUE THREAD

**Issue by lumen-novum** [cwwcww/prusa-mini-bowden-compensation#1]
Whenever I try to run the script, I get the following error message.
![image](https://github.com/cwwcww/prusa-mini-bowden-compensation/assets/91717010/d5b4e11c-9471-4ac9-a8df-353f80a6d90f)

It shouldn't be related to the file not having permission because that was a separate error message that I already fixed.

No error log was generated.
Script version 0.2.0
Mac OS 14.2

**Comment by cwwcww** [comment]
Could I ask you to export gcode from slicer without the script, run the script from terminal, passing the path to the exported gcode as the first argument, and post the output here?

**Comment by lumen-novum** [comment]
![image](https://github.com/cwwcww/prusa-mini-bowden-compensation/assets/91717010/a4ebe492-16b5-44a9-9dd0-a522d074a0e1)

No error log was generated and no new gcode file.

**Comment by cwwcww** [comment]
Apparently Mac OS requires the binaries to be signed to be executed, which I was not aware of. It will take me a while to figure out how and if I can do that properly. Sorry :(

For now, you can try seeking out some ways of bypassing that "Gatekeeper" security mechanism. [It looks like](https://openecoacoustics.org/resources/help-centre/software/unsigned/) you could add PrusaSlicer (and, possibly, terminal) as an exception, so it can run unsigned binaries. Please be aware of security implications before doing anything like this though.

**Comment by lumen-novum** [comment]
I tried making PrusaSlicer and Terminal an exception and it still gave me the same error without a log or gcode file... :(
![image](https://github.com/cwwcww/prusa-mini-bowden-compensation/assets/91717010/95f23200-0441-413c-83b4-728f4b89e9ba)

**Comment by lumen-novum** [comment]
I also tried the rest of the suggestions in the article and none of them worked... Are you sure Gatekeeper is the problem?

I temporarily disabled it system wide and it still gave me the same error message.

**Comment by cwwcww** [comment]
I'm not, to be honest - don't have experience with macs, and no access to one. I'll try to find a mac machine I could use to play around with it and fix this, but this wouldn't be fast.

I can suggest trying to run the script from terminal in a linux/windows vm. Far from practical, I know.

btw, when the script completes successfully, it replaces the initial gcode file, not creates a new one. I'm quite sure that the executable didn't even start in your case, but you can run it with no arguments just to be sure - if it is able to start, it would output something like "This executable is supposed to be executed by PrusaSlicer as a post-processing script.".

**Comment by lumen-novum** [comment]
TLDR: Install the correct architecture based on which homebrew architecture you have installed not your system architecture.

I found out why the script wasn't working. I have a M1 MacBook (ARM computer) so I thought that I needed to download the ARM version of your script. I installed x64 Homebrew (Mac OS package manager) a couple of weeks ago because I was modifying the MINI firmware which required x64 tools not ARM tools.

After trying the x64 version of your script, everything is working great. I'm pretty sure that's the reason it wasn't working. If you want to, I think it might be helpful to add a note to your README with what I put in my TLDR to help other developers avoid future headaches. Thank you for helping me investigate. 🙂

Side note- If someone else has ARM Homebrew installed on their system and would like to confirm my findings please feel free to do so.
